logstash-core/src/main/java/org/logstash/execution/QueueReadClientBase.java

```diff
@@ -148,11 +147,7 @@ public void rubyStartMetrics(final IRubyObject batch) {
      * @return Extracted queue batch
      */
     private static QueueBatch extractQueueBatch(final IRubyObject batch) {
-        if (batch instanceof JavaProxy) {
-            return (QueueBatch) ((JavaObject)batch.dataGetStruct()).getValue();
```

Comment:
seems like this was a (copy-pasta) bug

Reply:
I don't see any tests for this, maybe we should add some while at it, for one it should fail if that possible copy-pasta was invalid if not it will help assess we don't break expectations?

Reply:
thanks for checking - did not do that myself.
definitely will add a test here while we're at it.

Reply:
was wrong reading here - this actually worked as expected for the (removed) `JavaObject` wrapping
anyway, since we are removing the `JavaObject` wrapping here and these are LS internal classes 
-> the unwrapping now explains a proper Java proxy (will handle potential Ruby mocked sub-classes)